llvm/tools/llvm-exegesis/lib/MCInstrDescView.h

```diff
@@ -133,6 +133,12 @@ struct Instruction {
   // aliasing Use and Def registers.
   bool hasAliasingRegisters(const BitVector &ForbiddenRegisters) const;
 
+  // Whether this instruction is self aliasing through some registers.
+  // Repeating this instruction may execute sequentially by picking aliasing
+  // Def and Not Memory Use registers. It may also execute in parallel by
+  // picking non aliasing Def and Not Memory Use registers.
+  bool hasAliasingNotMemoryRegisters(const BitVector &ForbiddenRegisters) const;
```

Comment:
general question: it seems like you made quite some improvements on the infrastructure for memory instructions, which is nice. But could you add tests for these potentially corner cases so that we don't regress in the future.

Reply:
This function use to enable alternative strategy for instructions with memory operands. We have tests to cheek this behavior in `test/tools/llvm-exegesis/RISCV/latency-by-extension-A.s`